src/coreComponents/physicsSolvers/fluidFlow/StatOutputController.cpp

```diff
@@ -0,0 +1,181 @@
+  makeDirsForPath( m_outputDir );
+  string const filename = GEOS_FMT( "{}/{}", m_outputDir, outputManagerKey );
+
+  TimeHistoryOutput * timeHistory = &outputManager.registerGroup< TimeHistoryOutput >( outputManagerKey );
+  string_array & collectorPaths =  timeHistory->getReference< string_array >( TimeHistoryOutput::viewKeys::timeHistoryOutputTargetString() );
+  collectorPaths = packCollectionPaths;
+  string & outputFile =  timeHistory->getReference< string >( TimeHistoryOutput::viewKeys::timeHistoryOutputFilenameString() );
+  outputFile = filename;
+
+  m_timeHistories.push_back( timeHistory );
+}
+
+StatOutputController::StatOutputController( const string & name,
+                                            Group * const parent ):
+  TaskBase( name, parent ),
+  m_statistics( nullptr )
+{
+  registerWrapper( viewKeyStruct::outputDirString(), &m_outputDir ).
+    setApplyDefaultValue( "." ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Output directory for statistics HDF regions file" );
+}
+
+static std::set< string > const allowedChildTypes = { SinglePhaseStatistics::catalogName(),
+                                                      CompositionalMultiphaseStatistics::catalogName() };
+
+void StatOutputController::initializePreSubGroups()
+{
+
+  Group & problemManager = this->getGroupByPath( "/Problem" );
+  DomainPartition & domain = problemManager.getGroup< DomainPartition >( "domain" );
+  TasksManager & taskManager = this->getGroupByPath< TasksManager >( "/Tasks" );
+  OutputManager & outputManager = this->getGroupByPath< OutputManager >( "/Outputs" );
+
+  Group & meshBodies = domain.getMeshBodies();
+  std::vector< string > const groupNames = this->getSubGroupsNames();
+
+  GEOS_ERROR_IF( groupNames.size() != 1,
+                 GEOS_FMT( "{} must have exactly one of the following components : {}",
+                           getDataContext(), stringutilities::join( allowedChildTypes, "," ) ) );
+
+  m_statistics = &this->getGroup< TaskBase >( groupNames[0] );
+
+  // creating all needed components to read sub-groups statistics.
+  forSubStats( [&]( auto & statistics ) {
+    using STATSTYPE = typename TYPEOFREF( statistics );
+
+    statistics.getSolver()->forDiscretizationOnMeshTargets( meshBodies,
+                                                            [&] ( string const &,
+                                                                  MeshLevel & mesh,
+                                                                  arrayView1d< string const > const & regionNames )
+    {
+      ElementRegionManager & elemManager = mesh.getElemManager();
+      for( string const & regionName : regionNames )
+      {
+        ElementRegionBase & region = elemManager.getRegion( regionName );
+        string const regionStatPath = GEOS_FMT( "{}/{}", region.getPath(),
+                                                SinglePhaseStatistics::viewKeyStruct::regionStatisticsString() );
```

Comment:
You can move this `viewKey` in `FieldStatisticsBase` if you add a comment on it to say it only exists for statistics that are computed for regions.